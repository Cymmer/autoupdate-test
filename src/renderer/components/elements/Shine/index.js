import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

const Shine = ({ className }) => (
  <div className={cn(styles.Shine, className)} />
);

Shine.defaultProps = {
  className: null,
};

Shine.propTypes = {
  className: PropTypes.string,
};

export default Shine;
