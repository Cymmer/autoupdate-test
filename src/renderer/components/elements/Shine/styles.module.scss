@import '../../../scss/animations';
@import '../../../scss/variables';
@import '../../../scss/themes';

.Shine {
  height: 10px;
  width: 100%;
  border-radius: 16px;

  @include theme {
    background-color: t('shineBackgroundColor');
    background-image: t('shineBackgroundImage');
  }
  background-repeat: no-repeat;
  background-size: 800px 104px;
  display: inline-block;
  position: relative;

  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeholderShimmer;
  animation-timing-function: linear;
}
