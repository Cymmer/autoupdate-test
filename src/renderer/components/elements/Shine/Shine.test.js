import React from 'react';
import { shallow } from 'enzyme';
import Shine from './index';

describe('RENDERING - Shine', () => {
  it('should render with default props', () => {
    const component = shallow(<Shine />);
    expect(component).toMatchSnapshot();
  });
});
