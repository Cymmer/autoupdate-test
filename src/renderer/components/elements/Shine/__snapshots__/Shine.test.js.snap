// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Shine should render with default props 1`] = `
<div
  className="Shine"
/>
`;
